libclc/clc/lib/generic/math/clc_frexp.inc

```diff
@@ -26,7 +26,7 @@ __clc_frexp(__CLC_GENTYPE x, __CLC_ADDRESS_SPACE __CLC_INTN *ep) {
       (ai & (__CLC_INTN)MANTBITS_SP32);
 
   __CLC_INTN is_inf_nan_or_zero =
-      x == __CLC_FP_LIT(0.0) || __clc_isinf(x) || __clc_isnan(x);
+      ai == (__CLC_INTN)0 || __clc_isinf(x) || __clc_isnan(x);
```

Comment:
I know this was in the original code that we're essentially reverting to, but do we know this is correct for `-0.0`?

Reply:
Yes, this is correct for `-0.0`, `ai` is the bitcast of the absolute value of `x`.

Reply:
> Yes, this is correct for `-0.0`, `ai` is the bitcast of the absolute value of `x`.

Oh I'm silly, yes of course. Not sure how I thought the opposite was going on!